tests/setupAfterEnv.ts

```diff
@@ -0,0 +1,86 @@
+declare global {
+  // Using `namespace` here is okay because this is how the Jest types are
+  // defined.
+  /* eslint-disable-next-line @typescript-eslint/no-namespace */
+  namespace jest {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/ban-types
+    interface Matchers<R, T = {}> {
+      toNeverResolve(): Promise<R>;
+    }
+  }
+}
+
+const UNRESOLVED = Symbol('timedOut');
+// Store this in case it gets stubbed later
+const originalSetTimeout = global.setTimeout;
+const TIME_TO_WAIT_UNTIL_UNRESOLVED = 100;
+
+/**
+ * Produces a sort of dummy promise which can be used in conjunction with a
+ * "real" promise to determine whether the "real" promise was ever resolved. If
+ * the promise that is produced by this function resolves first, then the other
+ * one must be unresolved.
+ *
+ * @param duration - How long to wait before resolving the promise returned by
+ * this function.
+ * @returns A promise that resolves to a symbol.
+ */
+const treatUnresolvedAfter = (duration: number): Promise<typeof UNRESOLVED> => {
+  return new Promise((resolve) => {
+    originalSetTimeout(resolve, duration, UNRESOLVED);
+  });
+};
+
+expect.extend({
+  /**
+   * Tests that the given promise is never fulfilled or rejected past a certain
+   * amount of time (which is the default time that Jest tests wait before
+   * timing out as configured in the Jest configuration file).
+   *
+   * Inspired by <https://stackoverflow.com/a/68409467/260771>.
+   *
+   * @param promise - The promise to test.
+   * @returns The result of the matcher.
+   */
+  async toNeverResolve(promise: Promise<any>) {
```

Comment:
Hmm. Well, it never resolves at the point this matcher is called.

`toNotHaveResolved` sounds like it's saying the Promise hasn't resolved at that point in time. This isn't the same as that, this will actually `await` the Promise resolution, waiting as long as the timeout allows.

Reply:
For reference, this was copied from Elliot's eth-block-tracker tests: https://github.com/MetaMask/eth-block-tracker/blob/7c99edd010008c762bf870725f0e90a175c8fcf7/tests/setupAfterEnv.ts#L48

Reply:
👍